### PR TITLE
sink/mq(ticdc): gracefully close chann

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5655

### What is changed and how it works?
gracefully close chann. We must finish consuming the data here, otherwise it will cause the channel to not close properly.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Review by CharlesCheung96 — APPROVED** [review]

File: cdc/sink/mq/mq.go [hunk 1]
```diff
@@ -197,6 +197,15 @@ func (k *mqSink) FlushRowChangedEvents(
...
 // bgFlushTs flush resolvedTs to workers and flush the mqProducer
 func (k *mqSink) bgFlushTs(ctx context.Context) error {
+	defer func() {
+		go func() {
+			// We must finish consuming the data here,
+			// otherwise it will cause the channel to not close properly.
+			for range k.resolvedBuffer.Out() {
+				// Do nothing. We do not care about the data.
+			}
+		}()
```
Comment (overvenus) [inline cdc/sink/mq/mq.go:207]: Shall we wait the for loop done in the defer func?

Comment (hi-rustin) [reply cdc/sink/mq/mq.go:207]: No, closing is also done asynchronously, as long as we can return the error and the upper level calls close, we will always finish consuming the data and close the goruntine in chann normally.

Comment (overvenus) [reply cdc/sink/mq/mq.go:207]: What about moving the cleanup for loop to `mqSink.Close()`?

Comment (hi-rustin) [reply cdc/sink/mq/mq.go:207]: Good idea. Makes sense! Thanks

Comment (hi-rustin) [reply cdc/sink/mq/mq.go]: Done

File: cdc/sink/mq/mq.go [hunk 2]
```diff
@@ -347,6 +347,11 @@ func (k *mqSink) EmitDDLEvent(ctx context.Context, ddl *model.DDLEvent) error {
 // table sinks before closing it. So there is no writing after closing.
 func (k *mqSink) Close(ctx context.Context) error {
 	k.resolvedBuffer.Close()
+	// We must finish consuming the data here,
+	// otherwise it will cause the channel to not close properly.
+	for range k.resolvedBuffer.Out() {
```
Comment (asddongmen) [inline cdc/sink/mq/mq.go:352]: Is there a possible this for loop will block the `processorManager`.

**Review by overvenus — APPROVED** [review]
LGTM

File: cdc/sink/mq/mq.go (hunk 2 above)
Comment (hi-rustin) [reply cdc/sink/mq/mq.go:352]: We stop sending data before we stop sink. so I don't think so, and it doesn't make sense for the consumer to get stuck after closing?

**Comment by hi-rustin** [comment]
/hold

I will test it.

**Pull request merged**